### PR TITLE
DOC: update guideline for installation development environment using pip

### LINKED ISSUE #40595: BUG: Installation of the build dependencies fails
Following https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#creating-a-python-environment-pip to create dev environemnt
However, some failure pops up
```
ERROR: Command errored out with exit status 1:
     command: /Users/kaiqidong/pyvenv/pandas-dev/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/pip-install-a1dqlo0f/fastparquet_1298cd7d34ec45bc9cadbaacbc22c22c/setup.py'"'"'; __file__='"'"'/private/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/pip-install-a1dqlo0f/fastparquet_1298cd7d34ec45bc9cadbaacbc22c22c/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/pip-pip-egg-info-rsz4xmwi
         cwd: /private/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/pip-install-a1dqlo0f/fastparquet_1298cd7d34ec45bc9cadbaacbc22c22c/
    Complete output (95 lines):
    Processing numpy/random/_bounded_integers.pxd.in
    Processing numpy/random/_philox.pyx
    Traceback (most recent call last):
      File "/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/tools/cythonize.py", line 59, in process_pyx
        from Cython.Compiler.Version import version as cython_version
    ModuleNotFoundError: No module named 'Cython'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/tools/cythonize.py", line 235, in <module>
        main()
      File "/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/tools/cythonize.py", line 231, in main
        find_process_files(root_dir)
      File "/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/tools/cythonize.py", line 222, in find_process_files
        process(root_dir, fromfile, tofile, function, hash_db)
      File "/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/tools/cythonize.py", line 188, in process
        processor_function(fromfile, tofile)
      File "/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/tools/cythonize.py", line 64, in process_pyx
        raise OSError('Cython needs to be installed in Python as a module')
    OSError: Cython needs to be installed in Python as a module
    Running from numpy source directory.
    /var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/setup.py:485: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates
      run_build = parse_setuppy_commands()
    Traceback (most recent call last):
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/sandbox.py", line 154, in save_modules
        yield saved
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/sandbox.py", line 195, in setup_context
        yield
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/sandbox.py", line 250, in run_setup
        _execfile(setup_script, ns)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/sandbox.py", line 45, in _execfile
        exec(code, globals, locals)
      File "/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/setup.py", line 513, in <module>
      File "/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/setup.py", line 493, in setup_package
      File "/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/setup.py", line 290, in generate_cython
    RuntimeError: Running cythonize failed!
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/pip-install-a1dqlo0f/fastparquet_1298cd7d34ec45bc9cadbaacbc22c22c/setup.py", line 52, in <module>
        setup(
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/__init__.py", line 144, in setup
        _install_setup_requires(attrs)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/__init__.py", line 139, in _install_setup_requires
        dist.fetch_build_eggs(dist.setup_requires)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/dist.py", line 716, in fetch_build_eggs
        resolved_dists = pkg_resources.working_set.resolve(
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/pkg_resources/__init__.py", line 780, in resolve
        dist = best[req.key] = env.best_match(
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1065, in best_match
        return self.obtain(req, installer)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1077, in obtain
        return installer(requirement)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/dist.py", line 786, in fetch_build_egg
        return cmd.easy_install(req)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/command/easy_install.py", line 679, in easy_install
        return self.install_item(spec, dist.location, tmpdir, deps)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/command/easy_install.py", line 705, in install_item
        dists = self.install_eggs(spec, download, tmpdir)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/command/easy_install.py", line 890, in install_eggs
        return self.build_and_install(setup_script, setup_base)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/command/easy_install.py", line 1158, in build_and_install
        self.run_setup(setup_script, setup_base, args)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/command/easy_install.py", line 1144, in run_setup
        run_setup(setup_script, args)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/sandbox.py", line 253, in run_setup
        raise
      File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/contextlib.py", line 131, in __exit__
        self.gen.throw(type, value, traceback)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/sandbox.py", line 195, in setup_context
        yield
      File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/contextlib.py", line 131, in __exit__
        self.gen.throw(type, value, traceback)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/sandbox.py", line 166, in save_modules
        saved_exc.resume()
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/sandbox.py", line 141, in resume
        six.reraise(type, exc, self._tb)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/_vendor/six.py", line 685, in reraise
        raise value.with_traceback(tb)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/sandbox.py", line 154, in save_modules
        yield saved
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/sandbox.py", line 195, in setup_context
        yield
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/sandbox.py", line 250, in run_setup
        _execfile(setup_script, ns)
      File "/Users/kaiqidong/pyvenv/pandas-dev/lib/python3.8/site-packages/setuptools/sandbox.py", line 45, in _execfile
        exec(code, globals, locals)
      File "/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/setup.py", line 513, in <module>
      File "/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/setup.py", line 493, in setup_package
      File "/var/folders/sq/gc62mtns2n1gg_pgs6q2g0zw0000gn/T/easy_install-v2l4o5n8/numpy-1.20.1/setup.py", line 290, in generate_cython
    RuntimeError: Running cythonize failed!
    Cythonizing sources
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/58/49/dccb790fa17ab3fbf84a6b848050083c7a1899e9586000e34e3e4fbf5538/fastparquet-0.3.2.tar.gz#sha256=d81dcec444f4c4829234b9baac57c7125b8fbe9119c2eca2dee922650db49205 (from https://pypi.org/simple/fastparquet/) (requires-python:>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*,). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement fastparquet>=0.3.2
ERROR: No matching distribution found for fastparquet>=0.3.2
```

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [x] closes #40595 
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/development/contributing.rst [hunk 1]
```diff
@@ -325,7 +325,10 @@ Creating a Python environment (pip)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 If you aren't using conda for your development environment, follow these instructions.
-You'll need to have at least Python 3.6.1 installed on your system.
+You'll need to have at least Python 3.6.1 installed on your system. If your Python version
```
Comment (MarcoGorelli) [inline doc/source/development/contributing.rst:328]: 3.7 now?

Comment (charlesdong1991) [reply doc/source/development/contributing.rst]: Thanks! Updated!

Comment (MarcoGorelli) [reply doc/source/development/contributing.rst]: my bad, just noticed there's already #40552 open to address the 3.6.1 issue, sorry for my irrelevant comment

**Comment by jreback** [comment]
thanks @charlesdong1991

**Pull request merged**